e2e-tests/upgrade-consistency/run

```diff
@@ -8,6 +8,27 @@ test_dir=$(realpath $(dirname $0))
 
 cluster='some-name'
 
+wait_for_sts_generation() {
+    local name="${1}"
+    local desired_generation="${2}"
+    local size="${3}"
+    set +o xtrace
+    retry=0
+    until [[ "$(kubectl_bin get sts $name -o jsonpath='{.status.observedGeneration}')" == "${desired_generation}" \
```

Comment:
can we add addition check on CR.status.observedGeneration?
to make sure that operator procesed new CR

Reply:
sure

Reply:
Sorry, in this very test generation changes only once. So it's not a relevant check.